### PR TITLE
Finish implementing char validation

### PR DESCRIPTION
**Pull request by aochagavia**
The only thing missing right now are good integration tests (and maybe more descriptive error messages)

### CODE REVIEW

File: crates/ra_syntax/src/validation.rs [hunk 1]
```diff
@@ -42,15 +44,82 @@ fn validate_char(node: ast::Char, errors: &mut Vec<SyntaxError>) {
...
+                    errors.push(SyntaxError::new(UnclosedUnicodeEscape, range))
+                }
+
+                if code.len() == 0 {
+                    errors.push(SyntaxError::new(EmptyUnicodeEcape, range));
+                    return;
+                }
+
+                if code.len() > 6 {
+                    errors.push(SyntaxError::new(OverlongUnicodeEscape, range));
```
[the remark this answers is not in the document]
Comment (aochagavia) [reply crates/ra_syntax/src/validation.rs]: Good catch!

**Comment by aochagavia** [comment]
I think this is ready to be merged. I could make tests check for expected errors and their range (instead of checking if `errors > 0`), but I am not sure it is worth the hassle.

File: crates/ra_syntax/src/utils.rs [hunk 2]
```diff
@@ -78,3 +80,38 @@ pub(crate) fn validate_block_structure(root: SyntaxNodeRef) {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct MutAsciiString<'a> {
```
Comment (matklad) [inline crates/ra_syntax/src/utils.rs]: Let's use https://crates.io/crates/arrayvec crate instead?

**Pull request merged**